aten/src/ATen/native/mps/operations/Activation.mm

```diff
@@ -19,14 +19,15 @@
 Tensor relu_mps(const Tensor& self) {
   using namespace mps;
   using CachedGraph = MPSUnaryCachedGraph;
-  Tensor output = at::empty_like(self);
-  resize_tensor(&output);
-  TORCH_CHECK(output.is_mps());
-
   MPSGraphCache* cache_ = MPSGraphCache::getInstance();
 
   MPSStream* stream = getCurrentMPSStream();
 
+  bool executeGatherOp = !(self.is_contiguous(MemoryFormat::Contiguous) ||
+                           self.is_contiguous(MemoryFormat::ChannelsLast) ||
+                           self.is_contiguous(MemoryFormat::ChannelsLast3d));
+  Tensor output = at::empty_like(self, executeGatherOp ? MemoryFormat::Contiguous : MemoryFormat::Preserve);
```

Comment:
Why this logic is different than on line #85?

Reply:
Here output is allocated by us, so we can check if self is contiguous in any memory format, so output will also be contiguous in the same memory format and we can skip the gather completely since relu is performed element wise.

On line 85 output is same as self, so it has the same memory format (it's the same as empty_like(self, MemoryFormat::Preserve). For inplace op we'll be allocating a new contiguous output only if self is not contiguous in any mem format, then scatter it back to self.